### ISSUE TITLE
What is the difference between regression and bbox?

### ISSUE THREAD

**Issue by yxchng** [DaFuCoding/MTCNN_Caffe#12]

**Comment by DaFuCoding** [comment]
The "bbox" is a rectangle. Full name is "bounding box". But the "regression" is a offset scale about "bbox" like regressor.
Getting final face box by "bbox" and "regression". You can read "MTCNN::BoxRegress" to find details.

**Comment by yxchng** [comment]
Thanks for reply.

Just wondering, do you have any idea why the model is trained to produce both bounding box and regression (to finetune the box), rather than just produce a bounding box that is finetuned by the model itself? Shouldn't neural network be able to take care of that? Am I misunderstanding?

**Comment by DaFuCoding** [comment]
I think there are a few reasons as below.
Firstly, the pnet classification ability is weak, there are so many proposals to rnet so that to keep high recall rate. It need to reduce some proposals( IoU<threshold) but these proposals is part positive sample. So, the regression bounding box can aggregate proposals to relatively accurate position. This can increase classification accuracy and recall rate. I tried it by experiments.
Secondly, mtcnn don't like R-CNN and Fast RCNN detect in all image. Pnet like a slide window in image. Around face have so many proposals. NMS can't effect to remain positive proposals and reject negative proposals. So, the regression bounding box can more easy remain positive proposals in NMS.
Finally, the part sample is a trick about data augmentation and add some face context information so that to increase network ability in classification.
Hope to be of help to you.

**Comment by yxchng** [comment]
谢谢，还有一个疑问，为什么输入pnet的图要transpose？（ sample_single.t() )

**Comment by DaFuCoding** [comment]
因为原作者是利用matlab进行训练的，在matlab中加载图片与opencv加载图片方式不同，相差一个转置。并且caffe中默认卷积操作是按照行优先进行的，为了不修改caffe源码，所以进行了转置操作。
